VirtualReality/MRML/vtkVirtualRealityViewInteractorObserver.cxx

```diff
@@ -0,0 +1,294 @@
+// vtkInternal methods
+
+//---------------------------------------------------------------------------
+vtkVirtualRealityViewInteractorObserver::vtkInternal::vtkInternal(vtkVirtualRealityViewInteractorObserver* external)
+  : External(external)
+{
+  this->EventCallbackCommand = vtkCallbackCommand::New();
+  this->EventCallbackCommand->SetClientData(this->External);
+  this->EventCallbackCommand->SetCallback(vtkVirtualRealityViewInteractorObserver::CustomProcessEvents);
+}
+
+//---------------------------------------------------------------------------
+vtkVirtualRealityViewInteractorObserver::vtkInternal::~vtkInternal()
+{
+  this->EventCallbackCommand->Delete();
+}
+
+//---------------------------------------------------------------------------
+// vtkVirtualRealityViewInteractorObserver methods
+
+//----------------------------------------------------------------------------
+vtkVirtualRealityViewInteractorObserver::vtkVirtualRealityViewInteractorObserver()
+{
+  this->Internal = new vtkInternal(this);
+}
+
+//----------------------------------------------------------------------------
+vtkVirtualRealityViewInteractorObserver::~vtkVirtualRealityViewInteractorObserver()
+{
+  delete this->Internal;
+}
+
+//----------------------------------------------------------------------------
+void vtkVirtualRealityViewInteractorObserver::PrintSelf(ostream& os, vtkIndent indent)
+{
+  this->Superclass::PrintSelf(os,indent);
+}
+
+//----------------------------------------------------------------------------
+void vtkVirtualRealityViewInteractorObserver::SetInteractor(vtkRenderWindowInteractor *interactor)
+{
+  if (interactor == this->Interactor)
+    {
+    return;
+    }
+  // if we already have an Interactor then stop observing it
+  if (this->Interactor)
+    {
+    this->Interactor->RemoveObserver(this->Internal->EventCallbackCommand);
+    }
+
+  this->Superclass::SetInteractor(interactor);
+
+  if (interactor)
+    {
+    float priority = 0.0f;
+
+    /// 3D event bindings
+    // Move3DEvent: Already observed in vtkMRMLViewInteractorStyle
+    // Button3DEvent: Already observed in vtkMRMLViewInteractorStyle
+    interactor->AddObserver(vtkCommand::Menu3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::Select3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::NextPose3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::ViewerMovement3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::Pick3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::PositionProp3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::Clip3DEvent, this->Internal->EventCallbackCommand, priority);
+    interactor->AddObserver(vtkCommand::Elevation3DEvent, this->Internal->EventCallbackCommand, priority);
```

Comment:
We will update the base class in Slicer to take care of these. In the meantime, handling is explicitly done here.